File: 451-sort-characters-by-frequency/451-sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        


        cntr = Counter(s)

        heap = [(-v,k) for k ,v in cntr.items()]
        heapq.heapify(heap)

        res = [] 
        while heap:

            v, k = heapq.heappop(heap)
            res += [k]* -v 
        return ''.join(res)

File: 451-sort-characters-by-frequency/451-sort-characters-by-frequency.java
class Solution {
    public String frequencySort(String s) {
        
        
        HashMap<Character, Integer > cntr = new HashMap<>();

	for ( char c: s.toCharArray()) {

		cntr.put( c, cntr.getOrDefault(c,0) +1);
	}

	PriorityQueue<Character> heap = new PriorityQueue<>((a, b) -> (cntr.get(b) - cntr.get(a)));

	heap.addAll(cntr.keySet());

	StringBuilder sb = new StringBuilder();

	while (!heap.isEmpty()) {
		char c = heap.poll();
		for (int i =0 ; i < cntr.get(c) ; i++){
			sb.append(c);
		}
	}
	return sb.toString();
    }
}